test/e2e/app-dir/draft-mode/draft-mode.test.ts

```diff
@@ -0,0 +1,120 @@
+          expect($('#mode').text()).toBe('DISABLED')
+          expect($('#data').text()).toBe('')
+        })
+      }
+
+      it('should be disabled from api route handler', async () => {
+        const res = await next.fetch(`${basePath}state`)
+        expect(await res.text()).toBe('DISABLED')
+      })
+
+      it('should have set-cookie header on enable', async () => {
+        const res = await next.fetch(`${basePath}enable`)
+        const h = res.headers.get('set-cookie') || ''
+        Cookie = h.split(';').find((c) => c.startsWith('__prerender_bypass'))
+        expect(Cookie).toBeDefined()
+      })
+
+      it('should have set-cookie header with redirect location', async () => {
+        const res = await next.fetch(`${basePath}enable-and-redirect`, {
+          redirect: 'manual',
+        })
+        expect(res.status).toBe(307)
+        expect(res.headers.get('location')).toContain('/some-other-page')
+        const h = res.headers.get('set-cookie') || ''
+        const c = h.split(';').find((c) => c.startsWith('__prerender_bypass'))
+        expect(c).toBeDefined()
+      })
+
+      it('should genenerate rand when draft mode enabled', async () => {
+        const opts = { headers: { Cookie } }
+        const $ = await next.render$(basePath, {}, opts)
+        expect($('#mode').text()).toBe('ENABLED')
+        expect($('#rand').text()).not.toBe(origRandHome)
+      })
+
+      it('should read other cookies when draft mode enabled', async () => {
+        const opts = { headers: { Cookie: `${Cookie};data=cool` } }
+        const $ = await next.render$(`${basePath}with-cookies`, {}, opts)
+        expect($('#mode').text()).toBe('ENABLED')
+        expect($('#rand').text()).not.toBe(origRandWithCookies)
+        expect($('#data').text()).toBe('cool')
+      })
+
+      it('should be enabled from api route handler when draft mode enabled', async () => {
+        const opts = { headers: { Cookie } }
+        const res = await next.fetch(`${basePath}state`, opts)
+        expect(await res.text()).toBe('ENABLED')
+      })
+
+      it('should not perform full page navigation on router.refresh()', async () => {
+        const to = encodeURIComponent('/generate/foo')
+        const browser = await next.browser(
+          `${basePath}enable-and-redirect?to=${to}`
+        )
+        await browser.eval('window._test = 42')
+        await browser.elementById('refresh').click()
+
+        const start = Date.now()
+        while (Date.now() - start < 5000) {
+          const value = await browser.eval('window._test')
+          if (value !== 42) {
+            throw new Error('Detected a full page navigation')
+          }
+          await waitFor(200)
+        }
+
+        expect(await browser.eval('window._test')).toBe(42)
+      })
+    }
+
+    describe('in nodejs runtime', () => {
+      runTests({ basePath: '/' })
+    })
+
+    describe('in edge runtime', () => {
+      runTests({ basePath: '/with-edge/' })
+    })
```

Comment:
Nice job consolidating these with the `runTests()` helper!

In the future, you might want to keep separate files as entrypoints that import and call `runTests()`.

That way it makes it easier to run just one set of tests locally by `pnpm testheadless test/foo/file.test.js` running both with `pnpm testheadless test/foo`.

For example, see `sharp.test.ts` and `squoosh.test.ts` here

https://github.com/vercel/next.js/blob/0ecde6bd32dac474ec5342fc9e70c54363636717/test/integration/image-optimizer/test/sharp.test.ts#L26

https://github.com/vercel/next.js/blob/0ecde6bd32dac474ec5342fc9e70c54363636717/test/integration/image-optimizer/test/squoosh.test.ts#L8